sound/soc/sof/pcm.c

```diff
@@ -338,6 +338,13 @@ static snd_pcm_uframes_t sof_pcm_pointer(struct snd_pcm_substream *substream)
 	if (rtd->dai_link->no_pcm)
 		return 0;
 
+	/* return Xrun to ALSA for further handling */
+	if (spcm->stream[substream->stream].posn.xrun_size) {
+		/* only report once for each XRUN IPC */
+		spcm->stream[substream->stream].posn.xrun_size = 0;
+		return SNDRV_PCM_POS_XRUN;
```

Comment:
Not sure why this is done here, before pcm_pointer callback? Shouldn't this be done once all the information is reported, i.e. below line 350 (with the return removed).

Reply:
We have only one return value(position) with .pointer() callback, I aimed to report/return SNDRV_PCM_POS_XRUN once XRUN happened, which will let ALSA trigger stop->prepare->start later.